### DISCUSSION TITLE
"Use of manifest for determining app properties" on the CLI

### DISCUSSION THREAD

**Discussion by baroca** [filips123/PWAsForFirefox#377] [Q&A] [answered]
I'm trying to add a website (specifically Discord) without using the browser extension.
However, since Discord doesn't seem to have a manifest.json I'm unable to install via `firefoxpwa site install`.

I tried searching through the options available with the `help` but did not found an option similar to "Use of manifest for determining app properties" mentioned in the extension wiki.

Is it even possible or I'm trying something not intended to be used this way?
Should I provide an empty manifest myself?

Thanks for your work!

**Comment by filips123** [comment, marked as answer]
When "Use of manifest for determining app properties" is disabled, the extension will just generate a minimal manifest and send it to the connector as base64-encoded data URL. You can also provide such manifest yourself, altough it's probably a bit annoying to create and encode it.

For Discord specifically, you can use this manifest:

```json
{"start_url":"https://discord.com/","name":"Discord","description":"Discord is the easiest way to talk over voice, video, and text. Talk, chat, hang out, and stay close with your friends and communities.","icons":[{"src":"https://discord.com/assets/847541504914fd33810e70a0ea73177e.ico","type":null,"purpose":"any","sizes":""}]}
```

Which is then encoded into this manifest URL:

```
data:application/manifest+json;base64,eyJzdGFydF91cmwiOiJodHRwczovL2Rpc2NvcmQuY29tLyIsIm5hbWUiOiJEaXNjb3JkIiwiZGVzY3JpcHRpb24iOiJEaXNjb3JkIGlzIHRoZSBlYXNpZXN0IHdheSB0byB0YWxrIG92ZXIgdm9pY2UsIHZpZGVvLCBhbmQgdGV4dC4gVGFsaywgY2hhdCwgaGFuZyBvdXQsIGFuZCBzdGF5IGNsb3NlIHdpdGggeW91ciBmcmllbmRzIGFuZCBjb21tdW5pdGllcy4iLCJpY29ucyI6W3sic3JjIjoiaHR0cHM6Ly9kaXNjb3JkLmNvbS9hc3NldHMvODQ3NTQxNTA0OTE0ZmQzMzgxMGU3MGEwZWE3MzE3N2UuaWNvIiwidHlwZSI6bnVsbCwicHVycG9zZSI6ImFueSIsInNpemVzIjoiIn1dfQ==
```

So, you should be able to install it with `firefoxpwa site install DATA-URL --document-url https://discord.com/` (document URL needs to be manually provided).

Also, is there any specific reason why you don't use the extension to install Discord?

**Comment by baroca** [comment]
Thank you! It worked.

Regarding not using the extension, I was working on some scripts to automate apps installation.
